src/main/java/cubicchunks/lighting/LightingManager.java

```diff
@@ -45,26 +44,26 @@ public void columnSkylightUpdate(UpdateType type, Column column, int localX, int
 		int blockZ = Coords.localToBlock(column.getZ(), localZ);
 		switch (type) {
 			case IMMEDIATE:
-				Set<Integer> toDiffuse = SkyLightUpdateCubeSelector.getCubesY(column, localX, localZ, minY, maxY);
-				for (int cubeY : toDiffuse) {
-					boolean success = SkyLightCubeDiffuseCalculator.calculate(column, localX, localZ, cubeY);
+				IntSet toDiffuse = SkyLightUpdateCubeSelector.getCubesY(column, localX, localZ, minY, maxY);
+				for (IntCursor cubeY : toDiffuse) {
+					boolean success = SkyLightCubeDiffuseCalculator.calculate(column, localX, localZ, cubeY.value);
 					if (!success) {
-						queueDiffuseUpdate(column.getCube(cubeY), blockX, blockZ, minY, maxY);
+						queueDiffuseUpdate(column.getCube(cubeY.value), blockX, blockZ, minY, maxY);
 					}
 				}
 				break;
 			case QUEUED:
 				toDiffuse = SkyLightUpdateCubeSelector.getCubesY(column, localX, localZ, minY, maxY);
-				for (int cubeY : toDiffuse) {
-					queueDiffuseUpdate(column.getCube(cubeY), blockX, blockZ, minY, maxY);
+				for (IntCursor cubeY : toDiffuse) {
+					queueDiffuseUpdate(column.getCube(cubeY.value), blockX, blockZ, minY, maxY);
 				}
 				break;
 		}
 	}
 
 	public void tick() {
 		long timeStart = System.currentTimeMillis();
-		long timeStop = timeStart + TickBudget;
+		long timeStop = timeStart + CubicChunks.Config.getMaxTimeSpentGeneratingCubesPerTick();
```

Comment:
Ah, I thought they were the same when I asked you in chat, will change the name of the config option